src/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore/DatabaseErrorPageMiddleware.cs

```diff
@@ -216,15 +217,17 @@ void IObserver<DiagnosticListener>.OnNext(DiagnosticListener diagnosticListener)
         {
             switch (keyValuePair.Value)
             {
+                // NB: _localDiagnostic.Value can be null when this middleware has been leaked.
```

Comment:
can be null when DB operations take place outside of a request thread.

Reply:
Want me to add that?